src/diffusers/pipelines/self_segmentation_stable_diffusion/self_segmentation_stable_diffusion.py

```diff
@@ -0,0 +1,1012 @@
+# Copyright 2023 Self Segmentation Stable Diffusion Authors and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+import warnings
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import numpy as np
+import torch
+from packaging import version
+from transformers import CLIPImageProcessor, CLIPTextModel, CLIPTokenizer
+
+from diffusers.utils import (
+    deprecate,
+    is_accelerate_available,
+    is_accelerate_version,
+    is_nltk_available,
+    is_sklearn_available,
+    logging,
+    randn_tensor,
+    replace_example_docstring,
+)
+
+from ...configuration_utils import FrozenDict
+from ...image_processor import VaeImageProcessor
+from ...loaders import LoraLoaderMixin, TextualInversionLoaderMixin
+from ...models import AutoencoderKL, UNet2DConditionModel
+from ...models.attention_processor import Attention
+from ...pipelines.pipeline_utils import DiffusionPipeline
+from ...pipelines.stable_diffusion import StableDiffusionSafetyChecker
+from ...schedulers import KarrasDiffusionSchedulers
+from . import SelfSegmentationStableDiffusionPipelineOutput
+
+
+if is_sklearn_available():
+    from sklearn.cluster import KMeans, SpectralClustering
+if is_nltk_available():
+    import nltk
```

Comment:
what happens if one tries to run the pipeline with `nltk`? Does it error out? In this case let's maybe make sure to only import the pipeline if `nltk` is installed

Reply:
Agreed. For that to happen I think the check needs to go to `__init__.py` of the pipeline, no?